lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -30,13 +30,7 @@ function wp_theme_has_theme_json() {
 			return (bool) $theme_has_support;
 		}
 
-		// Has the own theme a theme.json?
-		$theme_has_support = is_readable( get_stylesheet_directory() . '/theme.json' ) ? 1 : 0;
-
-		// Look up the parent if the child does not have a theme.json.
-		if ( 0 === $theme_has_support ) {
-			$theme_has_support = is_readable( get_template_directory() . '/theme.json' ) ? 1 : 0;
-		}
+		$theme_has_support = is_readable( wp_get_theme()->get_file_path( 'theme.json' ) ) ? 1 : 0;
```

Comment:
There is only one (possibly irrelevant) edge case consideration here:
* `WP_Theme::get_file_path()` uses `file_exists()` to determine whether to use child theme or parent theme file.
* The logic here as well as `WP_Theme_JSON_Resolver::get_file_path_from_theme()` use `is_readable()` instead.

I don't know if this is going to be any real problem, but the two checks are different: It is technically possible for a file to exist but not be readable, and if that is the case the logic here now has changed.

Reply:
We can change that logic in core on the merge.

Reply:
I saw that we are deprecating the use of `wp_get_theme` at https://github.com/WordPress/gutenberg/pull/45770 (backports a core PR) because it's an expensive operation. At the same time, this PR and https://github.com/WordPress/gutenberg/pull/45955 proposes to add it to a few hotpaths (code we should optimize).

I'm having troubles following direction. Can you share more about _why_ we should remove it from some places and add it to others?

Reply:
What I'm hearing is that there are two issues:

- We don't cache things that should be cached (what [this issue](https://github.com/WordPress/gutenberg/issues/45171) is trying to fix, adding cache to public high-level methods such as [settings](https://github.com/WordPress/gutenberg/pull/45372), [stylesheet](https://github.com/WordPress/gutenberg/pull/45679), etc.). This is being worked on.
- Calling `WP_Theme` multiple times results in re-initializing, which is costly, so we avoid using `wp_get_theme`. Shouldn't we look at opportunities to improve the root cause here, such as making sure there's a single instance of the `WP_Theme` object, etc.? Is that something that anybody is looking at?

Reply:
> Calling WP_Theme multiple times results in re-initializing, which is costly, so we avoid using wp_get_theme. Shouldn't we look at opportunities to improve the root cause here, such as making sure there's a single instance of the WP_Theme object, etc.? Is that something that anybody is looking at?

There only way to do this, is some kind of global instance or static instance. There isn't a clear way of doing this. This is an issue that is out of scope of this PR and something that needs to handled in core.